src/dune/file_tree.ml

```diff
@@ -193,6 +193,14 @@ let readdir path =
     }
     |> Result.ok
 
+let get_vcs ~default:vcs ~path ~dirs =
```

Comment:
We need to look at both directories and files. For instance, git allows to have a `.git` file with a pointer to an external git directory.